components/movementsensor/rtkutils/ntrip.go

```diff
@@ -60,3 +92,113 @@ func NewNtripInfo(cfg *NtripConfig, logger logging.Logger) (*NtripInfo, error) {
 	logger.Debug("Returning n")
 	return n, nil
 }
+
+// ParseSourcetable gets the sourcetable and parses it.
+func (n *NtripInfo) ParseSourcetable() (*Sourcetable, error) {
+	reader, err := n.Client.GetSourcetable()
+	if err != nil {
+		return nil, err
+	}
+	defer func() {
+		if err := reader.Close(); err != nil {
+			log.Println("Error closing reader:", err)
+		}
+	}()
+
+	st := &Sourcetable{}
+	st.Streams = make([]Stream, 0, 200)
+	scanner := bufio.NewScanner(reader)
+	ln := ""
+
+Loop:
+	for scanner.Scan() {
+		ln := scanner.Text()
+		if strings.HasPrefix(ln, "#") || strings.HasPrefix(ln, "*") { // comment
+			continue
+		}
+		fields := strings.Split(ln, ";")
+		switch fields[0] {
+		case "CAS":
+			continue
+		case "NET":
+			continue
+		case "STR":
+			if str, err := ParseStream(ln); err == nil {
+				st.Streams = append(st.Streams, str)
+			}
+		case "ENDSOURCETABLE":
+			break Loop
+		default:
+			return nil, fmt.Errorf("%s: illegal sourcetable line: '%s'", n.URL, ln)
+		}
+	}
+
+	if ln != "ENDSOURCETABLE" {
+		return nil, fmt.Errorf("invalid sourcetable: missing string \"ENDSOURCETABLE\" in %s", ln)
+	}
+
+	return st, nil
+}
+
+// ParseStream parses a line from the sourcetable.
+func ParseStream(line string) (Stream, error) {
+	fields := strings.Split(line, ";")
+
+	if len(fields) < 19 {
+		return Stream{}, fmt.Errorf("missing fields at stream line: %s", line)
+	}
+
+	carrier, err := strconv.Atoi(fields[5])
+	if err != nil {
+		return Stream{}, fmt.Errorf("parsing the streams carrier in line: %s", line)
+	}
+
+	satSystems := strings.Split(fields[6], "+")
+
+	lat, err := strconv.ParseFloat(fields[9], 32)
+	if err != nil {
+		return Stream{}, fmt.Errorf("parsing the streams latitude in line: %s", line)
+	}
+	lon, err := strconv.ParseFloat(fields[10], 32)
+	if err != nil {
+		return Stream{}, fmt.Errorf("parsing the streams longitude in line: %s", line)
+	}
+
+	nmea, err := strconv.ParseBool(fields[11])
+	if err != nil {
+		return Stream{}, fmt.Errorf("parsing the streams nmea in line: %s", line)
+	}
+
+	sol, err := strconv.Atoi(fields[12])
+	if err != nil {
+		return Stream{}, fmt.Errorf("parsing the streams solution in line: %s", line)
+	}
+
+	fee := false
+	if fields[4] == "Y" {
+		fee = true
+	}
+
+	bitrate, err := strconv.Atoi(fields[17])
+	if err != nil {
+		bitrate = 0
+	}
+
+	return Stream{
+		MP: fields[1], Identifier: fields[2], Format: fields[3], FormatDetails: fields[4],
+		Carrier: carrier, NavSystem: satSystems, Network: fields[7], Country: fields[8],
+		Latitude: float32(lat), Longitude: float32(lon), Nmea: nmea, Solution: sol, Generator: fields[13],
+		Compression: fields[14], Authentication: fields[15], Fee: fee, BitRate: bitrate, Misc: fields[18],
+	}, nil
```

Comment:
if all of the field vairables are consistent we could give each index of the field a name  and return that in this structure, same as you've done with the named variables.